### DISCUSSION TITLE
Remote storage driver

### DISCUSSION THREAD

**Discussion by subwaymatch** [jupyterlite/jupyterlite#575] [Q&A] [answered]
### Question
Is there a way to store users' files into a remote storage (e.g., Firebase Storage, S3)? If not, is this something the Jupyterlite team is considering in the future? I'd love to contribute if that's the case.

### Use Case
My use case is to provide a Jupyter environment for students in Python courses, but I am looking to let them open their Jupyter notebooks from any machine.

I wasn't able to find any discussions of remote storage from the documentation.

From jupyterlite's ["Configuring" documentation page](https://jupyterlite.readthedocs.io/en/latest/configuring.html#local-storage-drivers).
![image](https://user-images.githubusercontent.com/1064036/159572825-10502c52-0456-4fe7-806e-3e768455f95a.png)

Thanks!

**Comment by jtpio** [comment, marked as answer]
Thanks @subwaymatch for opening this discussion.

> Is there a way to store users' files into a remote storage (e.g., Firebase Storage, S3)? If not, is this something the Jupyterlite team is considering in the future?

There isn't any remote storage option yet. But this should hopefully be feasible at some point at least via a third-party plugin, once the contents is made more extensible: https://github.com/jupyterlite/jupyterlite/issues/315

Linking to https://github.com/jupyterlite/jupyterlite/issues/545 where this is also discussed.

**Comment by subwaymatch** [comment]
@jtpio Thank you very much for the explanation and the links to related discussions! I'm thankful every day for Pyodide and Jupyterlite. 👍

**Comment by seigel** [comment]
Any update to this type of idea?

**Reply by JackYinpei** [reply]
Recently, I've been researching how to write my own program for storing files in remote storage. Here's a demo of a development plugin.
[demo to develop an extension](https://jupyterlab.readthedocs.io/en/latest/extension/extension_tutorial.html#extension-tutorial)  [describe of jupyterlab extension](https://jupyterlab.readthedocs.io/en/latest/extension/extension_dev.html)

 I just started too, so if you have any thoughts on starting development, we can brainstorm ideas together.

**Reply by seigel** [reply]
Awesome.  I'll take a look at what you did.  Just looking at being able to federate data in the notebook back to a rest end point
